service/src/main/java/greencity/service/EventServiceImpl.java

```diff
@@ -254,6 +257,21 @@ public EventVO findById(Long eventId) {
         return modelMapper.map(event, EventVO.class);
     }
 
+    @Override
+    @Transactional
+    public void disableEvent(Long eventId) {
+        //TODO: remove duplicated code
+        Set<EventAttenderDto> allEventAttenders = getAllEventAttenders(eventId);
+        UpdateEventDto eventUpdates = new UpdateEventDto();
+        eventUpdates.setId(eventId);
+        eventUpdates.setIsActive(false);
+        if(allEventAttenders.isEmpty()){
+            delete(eventId, (String) SecurityContextHolder.getContext().getAuthentication().getPrincipal());
+        } else {
+            update(eventUpdates, (String) SecurityContextHolder.getContext().getAuthentication().getPrincipal(), null);
+        }
```

Comment:
I'm not a fan of this approach. Consider moving the principal part into delete and update metods